model_analyzer/config/generate/model_variant_name_manager.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections.abc
+from copy import deepcopy
+from model_analyzer.constants import DEFAULT_CONFIG_PARAMS
+
+
+class ModelVariantNameManager:
+
+    def __init__(self):
+
+        # Stored as _variant_names[base_model_name][param_combo] = variant_name_string
+        self._variant_names = {}
+
+        # Stored as _model_name_index[base_model_name] = current_count_integer
+        self._model_name_index = {}
+
+    def get_model_variant_name(self, base_model_name, param_combo):
+        """
+        Given a base model name and a dict of parameters to be applied
+        to the base model config, return the name of the model variant
+
+        If the same input values are provided to this function multiple times, 
+        the same value should be returned
```

Comment:
Change to: "value **will** be returned"
There are no cases where this isn't true.

Reply:
fixed